src/System.Memory/src/System/SpanSortHelpers.Common.cs

```diff
@@ -0,0 +1,175 @@
+        }
+        //
+        // Type specific DirectComparer(s) to ensure optimal code-gen
+        //
+        internal struct SByteDirectComparer : IDirectComparer<sbyte>
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool GreaterThan(sbyte x, sbyte y) => x > y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThan(sbyte x, sbyte y) => x < y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThanEqual(sbyte x, sbyte y) => x <= y;
+        }
+        internal struct ByteDirectComparer : IDirectComparer<byte>
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool GreaterThan(byte x, byte y) => x > y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThan(byte x, byte y) => x < y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThanEqual(byte x, byte y) => x <= y;
+        }
+        internal struct Int16DirectComparer : IDirectComparer<short>
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool GreaterThan(short x, short y) => x > y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThan(short x, short y) => x < y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThanEqual(short x, short y) => x <= y;
+        }
+        internal struct UInt16DirectComparer : IDirectComparer<ushort>
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool GreaterThan(ushort x, ushort y) => x > y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThan(ushort x, ushort y) => x < y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThanEqual(ushort x, ushort y) => x <= y;
+        }
+        internal struct Int32DirectComparer : IDirectComparer<int>
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool GreaterThan(int x, int y) => x > y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThan(int x, int y) => x < y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThanEqual(int x, int y) => x <= y;
+        }
+        internal struct UInt32DirectComparer : IDirectComparer<uint>
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool GreaterThan(uint x, uint y) => x > y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThan(uint x, uint y) => x < y;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool LessThanEqual(uint x, uint y) => x <= y;
```

Comment:
nit: re-use GreaterThan, i.e. `!(x > y)`

Reply:
deleting this since not needed, but in general I would have kept this as is for optimal code-gen